share/wake/lib/core/list.wake

```diff
@@ -226,3 +226,28 @@ export def unzip = match _
   Pair a b, t =
     def Pair u v = unzip t
     Pair (a, u) (b, v)
+
+# Group a list by some comparison function.
+# Elements which compare equal are placed into the same bucket.
+# Buckets preserve the order of the original list.
+# The buckets appear in the sort-order specified by cmp.
+# Examples:
+#
+#   def cmp a b = a%3 < b%3
```

Comment:
Is it happenstance that we get `3` on both sides? What happens if they're different?

Reply:
If they are different, it will probably do something odd, since it's not a legal comparator.
(ie: a < b should imply !(b < a))

Reply:
I did consider taking two arguments; a project function and a compare function, so you would do this:
`groupBy (_/3) icmp (seq 10)`